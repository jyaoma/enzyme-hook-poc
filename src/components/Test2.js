import React, { useState, useCallback, useEffect } from 'react';
import axios from 'axios';

const Test2 = () => {
  const [catFacts, setCatFacts] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  const fetchCatFacts = useCallback(async () => {
    setIsLoading(true);
    try {
      const response = await axios.get('https://cat-fact.herokuapp.com/facts');
      setCatFacts(response.data.all);
    } catch (error) {
      console.error(error);
    } finally {
      setIsLoading(false);
    }
  });

  useEffect(() => {
    fetchCatFacts();
  }, [fetchCatFacts]);

  return <div className="cat-facts__container">
    {catFacts.map((catFact, i) => <div key={i}>{catFact.text}</div>)}
  </div>
}

export default Test2;
