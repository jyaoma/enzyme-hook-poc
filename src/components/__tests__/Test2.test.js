import React from 'react';
// import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import axios from 'axios';
import { act } from '@testing-library/react'

import Test2 from '../Test2';

jest.mock('axios');

describe('Test2', () => {
  it('should render correctly', () => {
    const rendered = renderer.create(<Test2 />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should start off empty', () => {
    axios.get.mockResolvedValue({data: {all: [{text:'The oldest cat video on YouTube dates back to 1894.'}]}})
    // let tree = mount(<Test2 />);
    // const container = tree.find('.cat-facts__container');

    // expect(container.props().children).toHaveLength(0);
  });
});