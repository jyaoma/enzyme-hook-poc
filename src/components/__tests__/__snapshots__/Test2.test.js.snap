// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test2 should render correctly 1`] = `
<div
  className="cat-facts__container"
/>
`;
